datafusion-cli/src/object_storage.rs

```diff
@@ -563,6 +563,592 @@ pub(crate) async fn get_object_store(
     Ok(store)
 }
 
+pub mod instrumented {
```

Comment:
I've put this into its own module because with this addition I feel like `object_storage` has probably grown past the point of being a single file. It seems like it should probably become a directory with 2 sub-modules, but this change set is already getting large enough I didn't want to go making large organizational changes prior to review.

Reply:
Yes, I agree that putting the instrumented object store into `object_storage/instrumented.rs` seems like a good idea